### ISSUE TITLE
class value type inheriting from two other aggregates is broken

### ISSUE THREAD

**Issue by Laeeth** [atilaneves/dpp#117]
obviously we don't have multiple inheritance in d.
but most important thing is to get the layout right.
currently we have:
``
class Currency: Convention<CurrencyId>, public relational_operators<Currency>
{}
```
translated as
```
struct Currency
{
  private:
    Convention!CurrencyId __base;
    alias __base this;
    relational_operators!Currency __base;
     alias __base this;
}
```
which is not going to get us very far.
so I don't know how multiple inheritance layout works.  but if you just stick them in order then number them __base1, __base2 etc and worry about no multiple alias this later.

**Comment by Laeeth** [comment]
Yes, but what I mean is that presuming you have the data layout right then it wouldn't matter for some purposes if there were no such thing as alias this.  Being able to represent the record layout correctly of c++ library is pretty valuable.

**Comment by Laeeth** [comment]
I know there's no multiple alias this in D - that's what I said.  If you get records right then can worry about the absence of multiple alias this later.  There's a decent chance that before dpp is complete we do have multiple alias this in the compiler.  If not, then if a class is used as a value type in another class then it certainly isn't going to work if you don't have record layout right...

**Comment by atilaneves** [comment]
I've had an idea about private members: Don't try to translate them at all. Instead, grab their size and alignment from libclang and declare a static array of that many bytes instead. In most real-world C++ code, fields are all private anyway.

**Comment by Laeeth** [comment]
For public members blacklisted or untranslatable you probably rather make it an Opaque(string name, size_t size)

**Comment by Laeeth** [comment]
If you do the above, fix operators, fix the fact the forward declaration problem, fix multiple inheritance (just number them) then I think quite a lot of c++ headers will be usable

**Comment by Laeeth** [comment]
Note that once dpp can compile other clang headers then many more things become easier.

**Comment by atilaneves** [comment]
I merged your PR into a branch of mine so that I can take it from there.
